src/net.cpp

```diff
@@ -472,7 +472,7 @@ CNode* CConnman::ConnectNode(CAddress addrConnect, const char *pszDest, bool fCo
         const std::vector<CService> resolved{Lookup(pszDest, default_port, fNameLookup && !HaveNameProxy(), 256)};
         if (!resolved.empty()) {
             const CService& rnd{resolved[GetRand(resolved.size())]};
-            addrConnect = CAddress{MaybeFlipIPv6toCJDNS(rnd), NODE_NONE};
+            addrConnect = CAddress{MaybeFlipIPv6toCJDNS(rnd), addrConnect.nServices};
```

Comment:
> either addrConnect or pszDest is provided by the caller

That used to be the case before this PR. This PR changes that to pass information via both params. I do not like this as I find it confusing but I did not bother mentioning it as too minor and subjective and I did not see a clear better way. Now that this has been raised already:

This PR uses `addrConnect` to pass the service flags even if the address to connect to is in `pszDest`. Can this be improved? Maybe instead of `pzsDest` pass `struct X { const char* dest, ServiceFlags services}`? And then pass `std::variant` of either `CAddress` or `X` (one argument instead of two).

Reply:
>  what is the "this" referring to?

Setting it to `addrConnect.nServices` instead of leaving the default initialized `NODE_NONE`.

> This PR changes that to pass information via both params. I do not like this

Ah, I didn't notice that. No strong opinion but in that case I think we should document the `pszDest` and `addrConnect ` arguments of `ConnectNode` and `OpenNetworkConnection`.

> adding a `use_v2transport` boolean argument to `ConnectNode` and `OpenNetworkConnection` instead

That sounds good.